lib/sensors/imd.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <io/can.hpp>
+
+namespace hyped::sensors {
+// Constants defined in IMD CAN reference manual
+constexpr std::uint8_t kRequestDataCanId            = 0xA100101;
+constexpr std::uint8_t kReturnDataCanId             = 0xA100100;
+constexpr std::uint8_t kRequestIsolationResistances = 0xE1;
+constexpr std::uint8_t kRequestIsolationState       = 0xE0;
+
+class Imd : public io::ICanProcessor {
+ public:
+  static std::optional<std::shared_ptr<Imd>> create(core::ILogger &logger,
+                                                    const std::shared_ptr<io::ICan> can);
+  Imd(core::ILogger &logger, const std::shared_ptr<io::ICan> can);
+
+  /**
+   * @brief Sends "Request Isolation resistances" CAN message to IMD
+   */
+  core::Result updateValues();
+
+  /**
+   * @brief Saves values returned from the IMD
+   */
+  core::Result processMessage(const io::CanFrame &frame);
+
+  std::uint16_t getRp();
+  std::uint16_t getRn();
+  std::uint8_t getIsolationState();
+
+ private:
+  core::ILogger &logger_;
+  std::shared_ptr<io::ICan> can_;
+  std::uint16_t rp_;
+  std::uint16_t rn_;
+  std::uint8_t iso_state_;
```

Comment:
naming

Reply:
Fair enough on _iso_state__
_rp__ and _rn__ came from the CAN reference manual, but I imagine code clarity is more important than consistency with a 3rd party manual

Reply:
in this case i think so. if ctrf+f for "resistance positive" doesn't turn anything up in the manual probably add a comment, but rp and rn are meaningless to everything outside this specific class so it is better to be clear